app/src/main/java/com/example/movieapp/crawler/InternetCall.kt

```diff
@@ -10,14 +21,59 @@ import androidx.lifecycle.MutableLiveData
 /***
  * Represent the type of movie list qe want for this call
  */
-enum class TypeList(s: String) {
+enum class TypeList(val s: String) {
     POPULAR("popular"),
     HIGHEST_RATE("top_rated")
 }
 
-private val URL = "http://api.themoviedb.org/3/movie/"
-private val KEY_API = "?api_key="
+private val TAG = "InternetCall"
 
-fun InternetCall(type:TypeList) {
 
+const val URL = "http://api.themoviedb.org/3/movie/"
+
+interface MoviesService {
+
+    @GET("{type}")
+    fun listOfMovies(
+        @Path("type") type: String,
+        @Query("api_key") s: String = KEY_PRIVATE
+    ): Call<AllData>
+}
+
+fun internetCall(type: TypeList): MutableLiveData<List<Movie>> {
+    val moviesRes = MutableLiveData<List<Movie>>()
+
+    val retrofit = Retrofit.Builder()
```

Comment:
Great! Using Retrofit and Gson will avoid a lot of pain :-)

Reply:
thanks a lot